### PR TITLE
1 Add Fr Localisation

### PR DESCRIPTION
**Pull request by Fr-Dae**
https://github.com/Rycieos/factorio-colorblind-ultimate/issues/7
https://github.com/Fr-Dae/factorio-colorblind-ultimate-dae/issues/1

:CH:
  Changes:
    - add Fr Localisation
    - add icon on desc
    - clean up of info.json
:CH:

### CODE REVIEW

File: info.json
Comment (Rycieos) [inline info.json:1]: Why change the formatting in this file? It was fine. I'll admit that it mixed 2 and 4 space indentation, but this makes the indentation worse in my opinion.

File: info.json
Comment (Rycieos) [inline info.json:10]: In general, I would prefer PRs to not bump version info. Not only might I want to add other changes to the release, but if this isn't merged before another release happens, then this PR would have a merge conflict.

File: info.json
Comment (Rycieos) [inline info.json:13]: Why remove the optional flag from this dependency? I realize that usually it is not possible to disable the base mod, but it is technically possible, and my mod is designed to be able to function without it.

File: changelog.txt
Comment (Rycieos) [inline changelog.txt:1]: I would prefer to update the changelog at release time.

File: locale/fr/locale.cfg
Comment (Rycieos) [inline locale/fr/locale.cfg:12]: This blank line is added by Crowdin, and will be added back if it is removed.

**Review by Rycieos — CHANGES_REQUESTED** [review]
Thanks for translating! I'm glad you did it through a PR instead of on Crowdin, as Crowdin is currently having issues and support has not been able to help me.

I am admittedly a very particular person when it comes to pull requests, so prepare for a big review. Before you read all my comments though, I want to lay out the 3 options we have to move forward, in my order of preference:

1. I have already applied all my suggestions myself, and created a commit that I am willing to merge. It is on my [pr/8](https://github.com/Rycieos/factorio-colorblind-ultimate/activity?ref=pr/8) branch. If you are willing to do so, you can take my commit into this PR, which I can then merge. The process for doing so would be `git fetch; git reset --hard origin/pr/8; git push -f`.
2. You can fix all my suggestions, and then request I re-review. This would obviously be slower than the first option.
3. You can tell me to just use my commit from option 1. The downside of this option is that GitHub will not mark this PR as merged, if you care about that.

So just let me know what path you want to take to move forward. And do not worry, I will be sure to credit you in the changelog when I write it.

**Comment by Fr-Dae** [comment]
> Thanks for translating! I'm glad you did it through a PR instead of on Crowdin, as Crowdin is currently having issues and support has not been able to help me.
> 
> I am admittedly a very particular person when it comes to pull requests, so prepare for a big review. Before you read all my comments though, I want to lay out the 3 options we have to move forward, in my order of preference:
> 
>     1. I have already applied all my suggestions myself, and created a commit that I am willing to merge. It is on my [pr/8](https://github.com/Rycieos/factorio-colorblind-ultimate/activity?ref=pr/8) branch. If you are willing to do so, you can take my commit into this PR, which I can then merge. The process for doing so would be `git fetch; git reset --hard origin/pr/8; git push -f`.
> 
>     2. You can fix all my suggestions, and then request I re-review. This would obviously be slower than the first option.
> 
>     3. You can tell me to just use my commit from option 1. The downside of this option is that GitHub will not mark this PR as merged, if you care about that.
> 
> 
> So just let me know what path you want to take to move forward. And do not worry, I will be sure to credit you in the changelog when I write it.

as a translator, crowdin is useful for automating long translations, but I find it more of a hassle than anything else, geany and deepl are more than enough. 
I use gitkraken, I create an issue, I create a branch from the issue and I how issue, commit and PR correctly as I was taught, I never use the terminal for tasks like this.
And when the PR is closed/merged, that closes the issue too, everything is clean and github's ‘autochangelog’ function is activated in the release (also called release update).

so it's clean.

I also thought about preparing the changelog and the json, it's an oversight that many contributors make. 
the basic function of fuisnner merge suits me, so I would be cited as a contributor, I like to improve my github statistics this way =)

File: info.json
Comment (Fr-Dae) [reply info.json:1]: the standard of json and factorio is based on 4 spaces, in some cases other indentation can break the code afterwards, and the devs of factorio are specialised in releasing updates without informing anyone of the consequences. 

the formatting is purely visual for ease of use, and so you can quickly find the version to modify at the bottom of the page, many of the mods I've edited/modified didn't even have a contact, think of it as a form of standardisation, there are several moderators and contributors who have launched the ‘pretty json’ trend.

File: info.json
Comment (Fr-Dae) [reply info.json:10]: It depends on the github, sometimes the devs like us to do a complete and careful job, others don't care, personally I avoid adding work to other people,

File: info.json
Comment (Fr-Dae) [reply info.json:13]: il n'est pas possible de lancer factorio sans factorio, donc lancer le mod sans la base de factorio est un non sense, de plus la standardisation de [base] implique de l'ecrire comme ceci.
techniquement cela devrai etre ? [base] mais je n'ai aucune idée de pourquoi les dev de factorio nous demande de l'ecrir ainsi.

File: changelog.txt
Comment (Fr-Dae) [reply changelog.txt:1]: same answer as before, contributors and translators are often asked to do a thorough and complete job, which also includes the changelog, 
but if you want to modify it or merge it with one of your own updates, that's up to you. 
Anyway, versioning on github doesn't matter, only mods.factorio.com counts, because that's what triggers update requests from players.

**Comment by Fr-Dae** [comment]
it's perfectly normal to review a code before merging it. i'd be surprised if you didn't!
have a nice day

**Comment by Fr-Dae** [comment]
iso8601 to factorio date format done,
resolve conversation if it's good for you , ready to merge for my side

**Review by Fr-Dae — COMMENTED** [review]
=)

File: info.json
Comment (Rycieos) [reply info.json:13]: Incorrect. See [this Reddit post from Klonan](https://github.com/Rycieos/factorio-colorblind-ultimate/discussions/6#discussioncomment-11199482):

> Base mod is a 'mod' in the context of Factorio, in that it adds or modifies prototype data.
>
> If you were to add all the required prototypes (There are some that the game can't run without) in another mod, then you could play without base mod enabled at all.

While I am not aware of any mod that currently does this, it is possible. And like I said, my mod is designed to be able to function without base activated, so there is no reason to force it to be.

File: info.json
Comment (Rycieos) [reply info.json:10]: Understood. And I can't be upset since I haven't added development guides to this project, so you have no way of knowing ahead of time. But I am telling you now. I have another change ready that I want to go in the same version, and it will not be a patch version bump.

File: info.json
Comment (Rycieos) [reply info.json:1]: > the standard of json ... is based on 4 spaces

Extremely incorrect. JSON is a very old standard, and it completely ignores whitespace. See the standard if you do not believe me: https://www.json.org/json-en.html

Whitespace in JSON is only for human readability.

> in some cases other indentation can break the code afterwards, and the devs of factorio are specialised in releasing updates without informing anyone of the consequences.

I disagree strongly. The devs of Factorio are, in my opinion, some of the best in the industry (and I would know, because I work in the industry). While they aren't perfect (they continually refuse to add colorblind accessibility support), they rarely break things on purpose. I would be shocked if they stopped accepting valid JSON. And even if you were right, the indentation style they would look for could just as easily be mine as yours.

> the formatting is purely visual for ease of use, and so you can quickly find the version to modify at the bottom of the page

That is a preference, and it is not my preference. I like this file in the format that it currently has.

**Review by Rycieos — CHANGES_REQUESTED** [review]
> iso8601 to factorio date format done

Why? ISO-8601 is a standard. Factorio does not have a "standard" date format, according to [the wiki](https://wiki.factorio.com/Tutorial:Mod_changelog_format#Date):

> There are no restrictions for the format of the date itself. 

You have picked the European style of date format. I could have picked the American style, but instead I picked ISO-8601 because **it is a standard**, and I also happen to think it is superior.

You are not likely to convince me to accept any of the changes outside of the translations. But even if you did, I never want PRs to contain unrelated changes. This PR is for a translation, and so it should **only** contain a translation.

So again, you have a choice. Either take my commit on my [pr/8](https://github.com/Rycieos/factorio-colorblind-ultimate/activity?ref=pr/8) branch, and force push to this PR, or I will merge that commit anyway.

**Comment by Fr-Dae** [comment]
> > iso8601 to factorio date format done
> 
> Why? ISO-8601 is a standard. Factorio does not have a "standard" date format, according to [the wiki](https://wiki.factorio.com/Tutorial:Mod_changelog_format#Date):
> 
> > There are no restrictions for the format of the date itself.

It's undoubtedly more a human concern than a regulatory one, but the modders I work for (especially on translations) all ask me to respect this format because it's written as it is in the documentation. Personally, I prefer the iso standard, at least it exists and is logical. 


> You are not likely to convince me to accept any of the changes outside of the translations. But even if you did, I never want PRs to contain unrelated changes. This PR is for a translation, and so it should **only** contain a translation.
a modification must always be recorded in a changelog. however, i agree with you about the iso standard. many people use the standard noted on the wiki, no idea why ¯\_(ツ)_/¯


> So again, you have a choice. Either take my commit on my [pr/8](https://github.com/Rycieos/factorio-colorblind-ultimate/activity?ref=pr/8) branch, and force push to this PR, or I will merge that commit anyway.
i don't have understand this part, btw

**Comment by Fr-Dae** [comment]
ok for 2 spaces, I know for example that tabs are not accepted. and generally a tab corresponds to 4 spaces, I think that's where it comes from. 


> The devs of Factorio are, in my opinion, some of the best in the industry 

What I mean by this is that the factorio devs regularly make updates, without prior announcement, that break or modify functions used by the community.
This is still the case with 2.0.16, which has suddenly broken mods that were working very well until then because a sub-subfunction has changed. 
and obviously the documentation wasn't up to date and the changelog was very vague on the subject.

**Comment by Rycieos** [comment]
> > The devs of Factorio are, in my opinion, some of the best in the industry
> 
> What I mean by this is that the factorio devs regularly make updates, without prior announcement, that break or modify functions used by the community. This is still the case with 2.0.16, which has suddenly broken mods that were working very well until then because a sub-subfunction has changed. and obviously the documentation wasn't up to date and the changelog was very vague on the subject.

I can agree with that to an extent. They changed `global.write_file()` to `helpers.write_file()`, but they did put that in the Changelog:

> Moved LuaGameScript::table_to_json, json_to_table, write_file, remove_path, direction_to_string, evaluate_expression, encode_string, decode_string, parse_map_exchange_string, check_prototype_translations, is_valid_sound_path, is_valid_sprite_path to LuaHelpers.

They also changed a ton of stuff about pollution (which broke my handling of it) and only mentioned it briefly:

> Added airborne-pollutant prototype and changed various pollution related properties to support multiple pollution types.

But again, this is JSON we are talking about. It is an accepted standard, and they are not going to stop accepting valid JSON. And definitely not as part of a minor or patch version bump.

> > So again, you have a choice. Either take my commit on my [pr/8](https://github.com/Rycieos/factorio-colorblind-ultimate/activity?ref=pr/8) branch, and force push to this PR, or I will merge that commit anyway.
> 
> i don't have understand this part, btw you need i target this branch for the PR and not my forked branch to main that it ? i have make a fork because i can't make branch here

Sorry, let me try again. I will not merge the 4 commits in this PR as they are now, as they include changes I do not want. I could merge commit 8f5aa7bf215ccfcac09b604584c76ad6ff037d85, but that would mark this PR as "Closed". What I think we would both rather want is for this PR to be marked as "Merged". For that to happen, I need you (because I cannot push to your branch) to update your branch to contain that commit.

In GitKraken it would work like this (I'm following [these docs](https://www.gitkraken.com/learn/git/git-reset#git-reset-gitkraken-client)):
1. Click Pull -> Fetch.
2. Checkout your `1-add-fr-localisation` branch.
3. Right click on commit `8f5aa7b` or branch `pr/8`, and click "Reset 1-add-fr-localisation to this commit" -> "Hard - discard all changes"
4. Click Push -> Force Push.

Or you can open the Terminal in GitKraken, and run these commands:
```bash
git fetch origin  # my remote might have a different name
git checkout 1-add-fr-localisation
git reset --hard 8f5aa7b
git push --force
```

**Comment by Fr-Dae** [comment]
https://github.com/Rycieos/factorio-colorblind-ultimate/issues/7
is now link to 
https://github.com/Rycieos/factorio-colorblind-ultimate/pull/9

**Pull request closed without merging**
